bob-common/src/metrics/exporter/send.rs

```diff
@@ -18,34 +18,58 @@ pub(super) async fn send_metrics(
     send_interval: Duration,
     prefix: String,
 ) {
-    let mut socket =
-        RetrySocket::new(address.parse().expect("Can't read address from String")).await;
+    let socket_sender = spawn_tcp_sender_task(address).await;
     let mut counters_map = HashMap::new();
     let mut gauges_map = HashMap::new();
     let mut times_map = HashMap::new();
-    let mut send_interval = interval(send_interval);
 
     loop {
-        send_interval.tick().await;
-        while let Ok(m) = timeout(METRICS_RECV_TIMEOUT, rx.recv()).await {
-            match m {
-                Some(Metric::Counter(counter)) => process_counter(&mut counters_map, counter),
-                Some(Metric::Gauge(gauge)) => process_gauge(&mut gauges_map, gauge),
-                Some(Metric::Time(time)) => process_time(&mut times_map, time),
-                // if recv returns None, then sender is dropped, then no more metrics would come
-                None => return,
+        let next_flush = Instant::now() + send_interval;
+        let mut current_time = Instant::now();
+        while current_time < next_flush {
+            if let Ok(m) = timeout(next_flush - current_time, rx.recv()).await {
+                match m {
+                    Some(Metric::Counter(counter)) => process_counter(&mut counters_map, counter),
+                    Some(Metric::Gauge(gauge)) => process_gauge(&mut gauges_map, gauge),
+                    Some(Metric::Time(time)) => process_time(&mut times_map, time),
+                    // if recv returns None, then sender is dropped, then no more metrics would come
+                    None => return,
+                }
             }
+            current_time = Instant::now();
         }
+        let mut res_string = String::new();
+        let ts = chrono::Local::now().timestamp();
+        flush_counters(&counters_map, &mut res_string, &prefix, ts).await;
+        flush_gauges(&gauges_map, &mut res_string, &prefix, ts).await;
+        flush_times(&mut times_map, &mut res_string, &prefix, ts).await;
+        if let Err(e) = socket_sender.send(res_string).await {
+            debug!("Can't send data to tcp sender task (reason: {})", e);
```

Comment:
I think it should be `warn` level. Because if sending failed there something serious happened with receiver side